src/sidecar/connections.ts

```diff
@@ -250,21 +252,11 @@ export async function waitForConnectionToBeStable(
     connectionStable.fire(id);
     // and trigger any kind of Topics/Schemas refresh to reenforce the error state / clear out any
     // old data
-    environmentChanged.fire(id);
+    environmentChanged.fire(id as unknown as EnvironmentId);
     logger.error(msg);
     return null;
   }
 
-  // per-type success actions ...
-  const type = connection.spec.type!;
-  if (type === ConnectionType.Direct) {
```

Comment:
Here are the main side-effects of `waitForConnectionToBeStable()` which migrated. Just Direct connections when the connection was deemed stable and not-null.